src/plugin.js

```diff
@@ -1,85 +1,64 @@
 'use strict';
-var Joi = require('joi');
+const Joi = require('joi');
+const _ = require('lodash');
 
-exports.register = function(plugin, options, next){
-  var schemaOptions =  Joi.alternatives().try(
-    Joi.object().keys({
-      lower_case: Joi.boolean().required(),
-      camel_case: Joi.boolean().when('lower_case', { is: true, then: Joi.invalid(true) }),
-      all_caps: Joi.boolean().when('lower_case', { is: true, then: Joi.invalid(true) }),
-      first_cap: Joi.boolean().when('lower_case', { is: true, then: Joi.invalid(true) }),
-      delete_original: Joi.boolean().optional()
-    }),
-    Joi.object().keys({
-      lower_case: Joi.boolean().when('camel_case', { is: true, then: Joi.invalid(true) }),
-      camel_case: Joi.boolean().required(),
-      all_caps: Joi.boolean().when('camel_case', { is: true, then: Joi.invalid(true) }),
-      first_cap: Joi.boolean().when('camel_case', { is: true, then: Joi.invalid(true) }),
-      delete_original: Joi.boolean().optional()
-    }),
-    Joi.object().keys({
-      lower_case: Joi.boolean().when('all_caps', { is: true, then: Joi.invalid(true) }),
-      camel_case: Joi.boolean().when('all_caps', { is: true, then: Joi.invalid(true) }),
-      all_caps: Joi.boolean().required(),
-      first_cap: Joi.boolean().when('all_caps', { is: true, then: Joi.invalid(true) }),
-      delete_original: Joi.boolean().optional()
-    }),
-    Joi.object().keys({
-      lower_case: Joi.boolean().when('first_cap', { is: true, then: Joi.invalid(true) }),
-      camel_case: Joi.boolean().when('first_cap', { is: true, then: Joi.invalid(true) }),
-      all_caps: Joi.boolean().when('first_cap', { is: true, then: Joi.invalid(true) }),
-      first_cap: Joi.boolean().required(),
-      delete_original: Joi.boolean().optional()
-    })
-  );
+const schemaOptions = 
+  Joi.object().keys({
+    lowerCase: Joi.boolean(),
+    schemaCamelCase: Joi.object(),
+    deleteOriginal: Joi.boolean().optional()
+  }).xor('lowerCase', 'schemaCamelCase');;
 
-  var validate = Joi.validate(options, schemaOptions);
+function generateKeyMapping(schema) {
+  const camelCaseSchema = _.mapKeys(schema, _.rearg(_.camelCase, 1));
```

Comment:
this looks like you're camel casing all the keys. If that's the case, then you should call the key `camelCase`

Reply:
otherwise just use the casing of the schema keys. See my overall review notes